src/main/java/org/openrewrite/quarkus/ChangeQuarkusPropertyKeyVisitor.java

```diff
@@ -0,0 +1,214 @@
+                    if (!profile.equals(this.profile)) {
+                        remainingProfiles.add(profile);
+                    }
+                }
+
+                // Remove the old property
+                tree = new org.openrewrite.properties.DeleteProperty(entry.getKey(), false)
+                        .getVisitor()
+                        .visit(tree, ctx);
+
+                // Add the new property for the named profile
+                String newKey = "%" + profile + "." + transformedKey;
+                tree = new org.openrewrite.properties.AddProperty(
+                        newKey,
+                        entry.getValue().getText(),
+                        null,
+                        entry.getDelimiter().getCharacter().toString()
+                ).getVisitor().visit(tree, ctx);
+
+                if (!remainingProfiles.isEmpty()) {
+                    // Add the new property for the named profile
+                    tree = new org.openrewrite.properties.AddProperty(
+                            "%" + String.join(",", remainingProfiles) + "." + keyWithoutProfile,
+                            entry.getValue().getText(),
+                            null,
+                            entry.getDelimiter().getCharacter().toString()
+                    ).getVisitor().visit(tree, ctx);
+                }
+            }
+        }
+
+        return tree;
+    }
+
+    private @Nullable Tree visitYamlDocuments(Tree tree, ExecutionContext ctx) {
+        Set<Yaml.Mapping.Entry> existingProperties = FindQuarkusProperties.find((Yaml.Documents) tree, oldPropertyKey, profile, changeAllProfiles);
+        for (Yaml.Mapping.Entry entry : existingProperties) {
+            String key = entry.getKey().getValue();
+            String keyWithoutProfile = getKeyWithoutProfile(key);
+            String transformedKey = replaceRegex(oldPropertyKey, newPropertyKey, keyWithoutProfile);
+            List<String> remainingProfiles = new ArrayList<>();
+            for (String profile : getProfiles(key)) {
+                if (!profile.equals(this.profile)) {
+                    remainingProfiles.add(profile);
+                }
+            }
+
+            // Remove the old property
+            tree = new org.openrewrite.yaml.DeleteProperty(key, false, false)
+                    .getVisitor()
+                    .visit(tree, ctx);
+
+            Cursor cursor = new Cursor(getCursor(), tree);
+
+            StringBuilder newProperties = new StringBuilder();
+            if (profile != null) {
+                formatKey(newProperties, transformedKey, entry.getValue().print(cursor), profile);
+            } else {
+                formatKey(newProperties, transformedKey, entry.getValue().print(cursor));
+            }
+
+            if (!remainingProfiles.isEmpty()) {
+                formatKey(newProperties, keyWithoutProfile, entry.getValue().print(cursor), remainingProfiles.toArray(new String[0]));
```

Comment:
With this change `void changeAllProfiles()` passes, but `void changeNamedProfile()` starts to fail due to too eager replacements there; might help hint at a proper fix, because this isn't it. :)
```suggestion
                formatKey(newProperties, transformedKey, entry.getValue().print(cursor), remainingProfiles.toArray(new String[0]));
```